src/autopas/utils/ThreeDimensionalMapping.h

```diff
@@ -52,12 +52,12 @@ constexpr T threeToOneD(const std::array<T, 3> &index3d, const std::array<T, 3>
  * @return The 3d index.
  */
 template <typename T>
-constexpr std::array<T, 3> oneToThreeD(T ind, const std::array<T, 3> &dims) {
+constexpr std::array<size_t, 3> oneToThreeD(T ind, const std::array<T, 3> &dims) {
   static_assert(std::is_integral<T>::value, "oneToThreeD requires integral types");
   std::array<T, 3> pos{};
-  pos[2] = ind / (dims[0] * dims[1]);
-  pos[1] = (ind - pos[2] * dims[0] * dims[1]) / dims[0];
-  pos[0] = ind - dims[0] * (pos[1] + dims[1] * pos[2]);
+  pos[2] = static_cast<size_t>(ind / (dims[0] * dims[1]));
+  pos[1] = static_cast<size_t>((ind % (dims[0] * dims[1])) / dims[0]);
+  pos[0] = static_cast<size_t>(ind % dims[0]);
```

Comment:
Now that I relook at it, the existing implementation is also correct, I didn't need to change. I just had some doubt so I changed it while debugging. But the problem was elsewhere. So, I think we can revert back or keep it, both are equivalent.